libs/ha/ha_common_libs_k8s.py

```diff
@@ -1441,48 +1443,77 @@ def get_node_resource_ids(node_obj, r_type, n_type=None, node_id=None):
+    def delete_kpod_with_shutdown_methods(self, master_node_obj, health_obj,
+                                          pod_prefix=None, kvalue=1,
+                                          down_method=common_const.RESTORE_SCALE_REPLICAS):
         """
-        Delete Single pod by setting replica set to 0. Check service status is in degraded mode.
+        Delete K pods by given shutdown method. Check and verify deleted/remaining pod's services
+        status, cluster status.
         :param master_node_obj: Master node object list
         :param health_obj: Health object
-        :param pod_prefix: Pod prefix to be deleted.
+        :param pod_prefix: Pod prefix to be deleted (Expected List type).
+        :param down_method: Pod shutdown/delete method.
+        :param kvalue: Number of pod to be shutdown/deleted.
         return : tuple
         """
-        LOGGER.info("Get pod name to be deleted")
-        pod_list = master_node_obj.get_all_pods(pod_prefix=pod_prefix)
-        pod_name = random.sample(pod_list, 1)[0]
-        hostname = master_node_obj.get_pod_hostname(pod_name=pod_name)
-
-        LOGGER.info("Deleting pod %s", pod_name)
-        resp = master_node_obj.create_pod_replicas(num_replica=0, pod_name=pod_name)
-        if resp[0]:
-            return False, f"Failed to delete pod {pod_name} by making replicas=0"
-        LOGGER.info("Successfully shutdown/deleted pod %s by making replicas=0", pod_name)
+        if pod_prefix is None:
+            pod_prefix = [common_const.POD_NAME_PREFIX]
+        pod_info = dict()
+        total_pod_type = [common_const.POD_NAME_PREFIX, common_const.SERVER_POD_NAME_PREFIX]
+        pod_data = {"method": None, "deployment_name": None, "deployment_backup": None,
+                    "hostname": None}
+        remaining = delete_pods = list()
+        for ptype in pod_prefix:
+            pod_list = master_node_obj.get_all_pods(pod_prefix=ptype)
+            # Get the list of Kvalue pods to be deleted for given pod_prefix list
+            delete_pods.extend(random.sample(pod_list, kvalue))
+
+        LOGGER.info("Get the list of all pods of total pod types.")
+        for ptype in total_pod_type:
+            remaining.extend(master_node_obj.get_all_pods(pod_prefix=ptype))
+
+        LOGGER.info("Delete %s by %s method", delete_pods, down_method)
+        for pod in delete_pods:
+            pod_info[pod] = pod_data.copy()
+            pod_info[pod]['hostname'] = master_node_obj.get_pod_hostname(pod_name=pod)
+            LOGGER.info("Deleting pod %s by %s method", pod, down_method)
+            if down_method == common_const.RESTORE_SCALE_REPLICAS:
+                resp = master_node_obj.create_pod_replicas(num_replica=0, pod_name=pod)
+                if resp[0]:
+                    # Failure here, required no pod restore in main test
+                    return False, f"Failed to delete pod {pod} by making replicas=0"
```

Comment:
If the failure occurs during deleting nth pod, are we taking care of recovering the previous failed pods. 
should that list be returned to test ?

Reply:
I don't see we are returning pod_info here.

Reply:
> I don't see we are returning pod_info here.

updated the same